### DISCUSSION TITLE
stroboscope effect

### DISCUSSION THREAD

**Discussion by MichaelZP** [keefo/NeewerLite#63] [Ideas]
Hello, 

First of all, thank you for creating and maintaining NeewerLite; it's been an incredible tool for Neewer light users! I have a feature suggestion that I believe could enhance its functionality.

#### Idea:
Add a stroboscope effect with the ability to define:
- **Blinking/pulsating frequency (Hz)**: Adjustable range, e.g., 1–100 Hz.
- **Duration**: Option for continuous or time-limited effect.

#### Use Case:
This feature would be useful for creative lighting setups, photography, video production, and even scientific experiments requiring precise light pulsation.

#### Question:
Would it be feasible to integrate this functionality into the current system? If not directly, could there be a workaround or suggestion on how users might achieve this effect?

Looking forward to hearing your thoughts!

**Comment by keefo** [comment]
Thank you for the suggestion! It’s a great creative idea, but after testing this during development, I've found it isn't feasible through a software-only approach for a few technical reasons:

Bluetooth Latency: Control commands sent over Bluetooth have a 'round-trip' delay. To achieve 100Hz, we would need to send a command every 5ms—which is faster than the Bluetooth protocol can reliably handle. This results in unstable flickering rather than a precise strobe.

Hardware Constraints: Most LED controllers in these lights aren't designed for high-frequency switching via external commands. In my tests, pushing high-frequency updates causes the light’s internal buffer to hang, making the device stop responding entirely.

Firmware vs. Software: For a true, high-speed stroboscope, the logic must live in the firmware (inside the light itself) so the hardware can pulse the LEDs at the electrical level.

Workaround: If you need a strobe effect, some Neewer models have a built-in 'Paparazzi' or 'Lightning' scene that you can trigger via NeewerLite, as those are handled directly by the light's internal hardware.

**Comment by keefo** [comment]
From my test I can make strobe effect at 3Hz safely.

**Comment by MichaelZP** [comment]
Thank you for the explanation. I understand that a software-only strobe
driven over BLE is not feasible because of latency and stability limits.

My question is slightly different: is there any possibility to implement
this function inside the device itself, at the firmware level, rather than
through external Bluetooth commands? In other words, could the light
generate the strobe pattern internally if the hardware is capable of it, or
is this also blocked by the controller design?

I’m asking because I understand BLE is too slow for a precise 100 Hz
effect, but perhaps an internal firmware-based solution could still be
possible.


MZP

niedz., 12 kwi 2026, 10:49 użytkownik Xu Lian ***@***.***>
napisał:

**Comment by keefo** [comment]
I would say only Neewer has firmware source code, so only them can implement it. Not possible from this app.
